src/services/jsTyping.ts

```diff
@@ -6,40 +6,81 @@
 /* @internal */
 namespace ts.JsTyping {
 
+    type AutoTypingsInfo = {
+        _autoTypings: string[],
+        autoTypingsEnabled: boolean,
+        include: string[],
+        exclude: string[]
+    };
+
     type HostType = {
+        directoryExists: (fileName: string) => boolean;
         fileExists: (fileName: string) => boolean;
         readFile: (path: string, encoding?: string) => string;
         readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
         writeFile: (path: string, data: string, writeByteOrderMark?: boolean) => void;
     };
 
-    interface TsdJsonCacheInfo {
-        path: string;
-        value: any;
-    }
-
     var _host: HostType;
+
+    var autoTypingsComment =
+        `//  This file contains a list of JavaScript typings (d.ts) that have
+//  been auto-referenced to provide a better completion list experience.
+//  For more information see the following page on the TypeScript wiki:
+//  https://github.com/Microsoft/TypeScript/wiki/AutoTypings
+`;
+
+    var autoTypingsDisabled = `The autoTypingsEnabled property is disabled for this project, no typings are being auto-referenced.
+`;
     
     // a typing name to typing file path mapping
     var inferredTypings: Map<string> = {};
 
-    function tryParseJson(jsonPath: string) {
-        try {
-            return JSON.parse(jsonPath);
+    function tryParseJson(host: HostType, jsonPath: string): any {
+        if (host.fileExists(jsonPath)) {

```

Comment:
As the `_host` has a global references, no need to pass it as a parameter anymore